actors/hierarchical_sca/src/state.rs

```
@@ -112,32 +94,20 @@ impl State {
         if val < self.min_stake {
             return Err(anyhow!("call to register doesn't include enough funds"));
         }
-        let mut subnets =
-            make_map_with_root_and_bitwidth::<_, Subnet>(&self.subnets, rt.store(), HAMT_BIT_WIDTH)
-                .map_err(|e| {
-                    e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to load subnets")
-                })?;
-
-        let empty_topdown_array =
-            Array::<(), BS>::new_with_bit_width(rt.store(), CROSSMSG_AMT_BITWIDTH)
-                .flush()
-                .map_err(|e| anyhow!("Failed to create empty messages array: {}", e))?;
-
-        let subnet = Subnet {
-            id: id.clone(),
-            stake: val,
-            top_down_msgs: empty_topdown_array,
-            circ_supply: TokenAmount::zero(),
-            status: Status::Active,
-            nonce: 0,
-            prev_checkpoint: Checkpoint::default(),
-        };
-        set_subnet(&mut subnets, &id, subnet)?;
-        self.subnets = subnets.flush().map_err(|e| {
-            e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to flush subnets")
-        })?;
+        // XXX: Is the update going to fail if the subnet already exist?
```

I reckon it won't, but I don't know if there's a check anywhere to disallow calling the actor with an already existing `id`.

---

I meant, can I trigger the registration twice from the CLI for example? I peeked into `set` called in `subnets.set(id.to_bytes().into(), subnet)` and it says:
```
    /// If the HAMT did not have this key present, `None` is returned.
    ///
    /// If the HAMT did have this key present, the value is updated, and the old
    /// value is returned. The key is not updated, though;
```
That sounds to me like we will might increment the number of subnets with `self.total_subnets += 1;` but then just reset the state of an existing subnet. 

Or is there are place where this check is already happening? Perhaps here we should return an error if the return value from `set` is not `None`?